x-pack/plugins/spaces/server/lib/saved_objects_client/spaces_saved_objects_client.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class SpacesSavedObjectsClient {
+  constructor(options) {
+    const {
+      request,
+      baseClient,
+      spaceUrlContext,
+    } = options;
+
+    this.errors = baseClient.errors;
+
+    this._client = baseClient;
+    this._request = request;
+
+    this._spaceUrlContext = spaceUrlContext;
+  }
+
+  async create(type, attributes = {}, options = {}) {
+
+    if (this._isTypeSpaceAware(type)) {
+      options.extraBodyProperties = {
+        ...options.extraBodyProperties,
+        spaceId: await this._getSpaceId()
+      };
+    }
+
+    return await this._client.create(type, attributes, options);
+  }
+
+  async bulkCreate(objects, options = {}) {
+    options.extraBodyProperties = {
+      ...options.extraBodyProperties,
+      spaceId: await this._getSpaceId()
+    };
+
+    return await this._client.bulkCreate(objects, options);
+  }
+
+  async delete(type, id) {
+    return await this._client.delete(type, id);
+  }
+
+  async find(options = {}) {
+    const spaceOptions = {};
+
+    if (this._isTypeSpaceAware(options.type)) {
+      const spaceId = await this._getSpaceId();
+
+      spaceOptions.queryDecorator = (query) => {
```

Comment:
Instead of using the decorator here, could we pass in the additional filter and let the SavedObjectsClient know how to append it? It's less generic, but let's us be less concerned with the internal implementation of the query itself.